### PR TITLE
👔 Improve recommendation display items & label

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2024-09-20 凌晨1 06 56
截圖 2024-09-20 凌晨1 08 56

### CODE REVIEW

File: src/components/NFTMessage/Identity.vue [hunk 1]
```diff
@@ -92,6 +96,9 @@ export default {
       });
     },
     userLabel() {
+      if (this.customUserLabel) {
+        return this.customUserLabel;
+      }
```
Comment (williamchong) [inline src/components/NFTMessage/Identity.vue:101]: Can we just pass a new type `publisher` and change all `this.type === 'creator'` logic to `creator || publisher` accordingly?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/components/NFTMessage/Identity.vue (hunk 1 above)
Comment (AuroraHuang22) [reply src/components/NFTMessage/Identity.vue:107]: [🎨 Use type props to identify user label](https://github.com/likecoin/liker-land/pull/1821/commits/341180c2ae62d87f4b8bc00067a275a98659b4e1)

File: src/components/NFTMessage/Identity.vue [hunk 2]
```diff
@@ -92,11 +96,15 @@ export default {
       });
     },
     userLabel() {
-      return this.$t(
-        this.type === 'creator'
-          ? 'identity_type_creator'
-          : 'identity_type_collector'
-      );
+      switch (this.type) {
```
Comment (williamchong) [inline src/components/NFTMessage/Identity.vue:99]: Please also update line 95

File: src/components/NFTMessage/Identity.vue [hunk 3]
```diff
@@ -78,6 +78,10 @@ export default {
       type: Number,
       default: 42,
     },
+    customUserLabel: {
+      type: String,
+      default: undefined,
+    },
```
Comment (williamchong) [inline src/components/NFTMessage/Identity.vue:84]: Please remove

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**